src/main/java/com/labsynch/labseer/service/LotServiceImpl.java

```diff
@@ -80,52 +87,107 @@ public Lot updateLotWeight(Lot lot) {
 		return Lot.findLot(lot.getId());
 	}
 
+	public Integer predictReparentLotNumber(Parent parent, Lot lot) {
+		// With a lot and parent, predict what the lot number would be
+		Integer newLotNumber;
+		if(propertiesUtilService.getMaxAutoLotNumber() != null && lot.getLotNumber() > propertiesUtilService.getMaxAutoLotNumber()){
+			newLotNumber = lot.getLotNumber();
+		} else {
+			newLotNumber = Lot.getMaxParentLotNumber(parent) + 1;
+		}
+		return newLotNumber;
+	}
+
 	@Override
 	@Transactional
-	public Lot reparentLot(String lotCorpName, String parentCorpName, String modifiedByUser) {
+	public ReparentLotResponseDTO reparentLot(String lotCorpName, String parentCorpName, String modifiedByUser, Boolean updateInventory, Boolean updateAssayData, Boolean dryRun) throws DupeLotException{
 
-		// incoming lot
-		// name of new adoptive parent
-		// logic to deal with lot numbering -- just increment existing lot number of
-		// adoptive parent
-		// note -- need to refactor to deal with more complicated parent/SaltForm/lot
-		// setups versus simpler parent/lot
+		logger.info("Got request to reparent lot: " + lotCorpName + " to parent: " + parentCorpName);
+		
+		ReparentLotResponseDTO reparentLotResponseDTO = new ReparentLotResponseDTO();
+		reparentLotResponseDTO.setOriginalLotCorpName(lotCorpName);
+		reparentLotResponseDTO.setModifiedBy(modifiedByUser);
+
+		Lot queryLot = Lot.findLotsByCorpNameEquals(lotCorpName).getSingleResult();
+		reparentLotResponseDTO.setOriginalLotNumber(queryLot.getLotNumber());
 
+		SaltForm saltForm = queryLot.getSaltForm();
 		Parent adoptiveParent = Parent.findParentsByCorpNameEquals(parentCorpName).getSingleResult();
+		Set isoSalts = saltForm.getIsoSalts();
+		isoSalts.size();
+		if(dryRun) {
+			Lot.entityManager().detach(queryLot);
```

Comment:
If dry run, we need to detach the objects so their data doesn't get persisted automatically by hibernate.

Reply:
I added some comments now.